pkg/tarmak/cluster/cluster.go

```diff
@@ -142,27 +146,30 @@ func validateClusterTypes(poolMap map[string][]*clusterv1alpha1.InstancePool, cl
 		result = multierror.Append(result, fmt.Errorf("a %s needs to have more than one '%s' server pool", clusterType, clusterv1alpha1.InstancePoolTypeMaster))
 	}
 
-	return result
+	return result.ErrorOrNil()
 }
 
 // validate server pools
-func (c *Cluster) validateInstancePools() (result error) {
+func (c *Cluster) validateInstancePools() error {
+	var result *multierror.Error
+
 	for _, instancePool := range c.InstancePools() {
 		err := instancePool.Validate()
 		if err != nil {
 			result = multierror.Append(result, err)
 		}
 	}
-	return result
-}
 
-// Verify cluster
-func (c *Cluster) Verify() (result error) {
-	return c.VerifyInstancePools()
+	if err := c.Environment().Provider().ValidateInstanceTypes(c.InstancePools()); err != nil {
```

Comment:
I think this needs to become a verify check and not an validate check. This will contact AWS and is a more expensive call. This is to keep working in line with https://github.com/jetstack/tarmak/issues/280

Reply:
done